src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy, ...extraData,
+        };
```

Comment:
```suggestion
        stateCopy = {
          ...stateCopy,
          ...extraData,
        };
```
By the way, it is better to use Object.assign to add extra data.
In that case, stateCopy can be a const
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign